src/assets/js/react/components/Help/HelpContainer.js

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import request from 'superagent'
+import { updateResult, deleteResult } from '../../actions/help'
+
+
+export const doHandleChange = data => ({
+  searchInput: data
+});
+
+export const toggleLoadingTrue = () => ({
+  loading: true
+});
+
+export const toggleLoadingFalse = () => ({
+  loading: false
+})
+
+export const fetchData = searchInput => {
+  // Request API call
+  return request.get(`https://gravitypdf.com/wp-json/wp/v2/v5_docs/?search=${searchInput}`);
+}
+
+
+class HelpContainer extends Component {
+
+  // Initialize component state
+  state = {
+    searchInput: '',
+    loading: false
+  }
+
+  onHandleChange = e => {
+    // Set loading to true
+    this.setState(doHandleChange(e.target.value));
+    // Set searchInput state value
+    this.searchInputLength(e.target.value);
+  }
+
+  searchInputLength = data => {
+    if (data.length > 3) {
+      // Set loading to true
+      this.setState(toggleLoadingTrue);
+      // Request API call
+      fetchData(data)
```

Comment:
Let's swap to async / await for any asynchronous tasks. You can see an example of this here https://github.com/GravityPDF/gravity-pdf/blob/78e5a2138df1170871100a8a8c9116d796505ae5/src/assets/js/react/components/CoreFonts/CoreFontContainer.js#L105

Reply:
I think the current code is much better so we can import the same function and run it into the unit testing.

Reply:
Fixed - swapped to use async / await.